Product/Production/Adapters/General/CONNECTAdminGUI/src/main/java/gov/hhs/fha/nhinc/admingui/managed/CertficateBean.java

```diff
@@ -198,104 +203,146 @@ public void onAliasValueEdit(CellEditEvent event) {
         }
     }
 
-    public void openTrustStorePasskeyDlgForDelete() {
+    public void openTrustStorePasskeyDlgForDelete() throws CertificateManagerException {
         if (selectedTSCertificate == null) {
             HelperUtil.addMessageError(TRUST_STORE_MSG, "Select a certificate to delete");
-        } else if (!isVerifiedTrustStoreUser()) {
+        } else if (isHashTokenEmpty()) {
             RequestContext.getCurrentInstance().execute("PF('deletePassKeyDlg').show();");
         } else {
-            deleteCertificate();
+            validateAndDeleteCertificate();
+        }
+    }
+
+    public void openTrustStorePasskeyDlgForUpdate() throws CertificateManagerException {
+        if (selectedTSCertificate == null) {
+            HelperUtil.addMessageError(TRUST_STORE_MSG, "Select a certificate to update");
+        } else if (isHashTokenEmpty()) {
+            RequestContext.getCurrentInstance().execute("PF('editPassKeyDlg').show();");
+        } else {
+            validateAndUpdateCertificate();
         }
     }
 
     public void openTrustStorePasskeyDlgForImport() throws CertificateManagerException {
         if (selectedCertificate == null) {
-            HelperUtil.addMessageError(IMPORT_PASS_KEY_ERR_MSG, "Select certificate for import");
+            HelperUtil.addMessageError(IMPORT_PASS_ERR_MSG_ID, "Select certificate for import");
         } else if (StringUtils.isBlank(selectedCertificate.getAlias()) || ALIAS_PLACEHOLDER.equals(selectedCertificate.
-                getAlias())) {
-            HelperUtil.addMessageError(IMPORT_PASS_KEY_ERR_MSG, "Enter an alias for the certificate");
-        } else if (!isVerifiedTrustStoreUser()) {
+            getAlias())) {
+            HelperUtil.addMessageError(IMPORT_PASS_ERR_MSG_ID, "Enter an alias for the certificate");
+        } else if (isHashTokenEmpty()) {
             RequestContext.getCurrentInstance().execute("PF('importPassKeyDlg').show();");
         } else {
-            importSelectedCertificate();
+            validateAndImportCertificate();
         }
     }
 
-    public void validateAndDeleteCertificate() {
-        if (service.validateTrustStorePassKey(trustStorePasskey)) {
-            if (isRememberMe()) {
-                HelperUtil.getHttpSession(false).setAttribute(VERIFIED_TRUSTSTORE_USER, true);
+    public void validateAndDeleteCertificate() throws CertificateManagerException {
+        if (isHashTokenEmpty()) {
+            String hashToken = service.getHashToken(trustStorePasskey);
+            if (service.validateTrustStorePassKey(trustStorePasskey)) {
```

Comment:
we should not validate the hashtoken on client since we don't know the system password.